source/applications/advanced/gamma_correction.py

```diff
@@ -0,0 +1,89 @@
+"""
+Capture a 2D image and apply a configurable gamma correction.
+"""
+
+import argparse
+import numpy as np
+import cv2
+import zivid
+
+
+def _options():
+    """Configure and take command line arguments from user
+
+    Returns:
+        Arguments from user
+
+    """
+    parser = argparse.ArgumentParser(
+        description=(
+            "Capture 2D image and apply gamma correction\n"
+            "Example:\n\t $ python gamma_correction.py 2"
+        ),
+        formatter_class=argparse.RawTextHelpFormatter,
+    )
+
+    parser.add_argument(
+        "gamma", nargs="?", type=float, help="Gamma correction value", default="2"
+    )
+
+    return parser.parse_args()
+
+
+def adjust_gamma(image, gamma: float):
+    """Adjust gamma on image.
+
+    Args:
+        image: OpenCV image to be corrected
+        gamma: Gamma setting to apply
+
+    Returns:
+        Gamma adjusted (OpenCV) image
+
+    """
+    # build a lookup table mapping the pixel values [0, 255] to
+    # their adjusted gamma values
+    inv_gamma = 1.0 / gamma
+    table = np.array(
+        [((i / 255.0) ** inv_gamma) * 255 for i in np.arange(0, 256)]
+    ).astype("uint8")
+    # apply gamma correction using the lookup table
+    return cv2.LUT(image, table)
+
+
+def _capture_2d_image(camera):
+    """Capture and extract 2D image, then convert from RGBA and return BGR.
+
+    Args:
+        camera: Zivid Camera handle
+
+    Returns:
+        OpenCV BGR image
+
+    """
+    settings_2d = zivid.Settings2D()
+    with camera.capture_2d(settings_2d) as frame_2d:
+        image = frame_2d.image()
+        image_array = image.to_array()
+        return np.dstack([image_array["b"], image_array["g"], image_array["r"]])
+
+
+def _main():
+    app = zivid.Application()
+
+    user_options = _options()
+
+    camera = app.connect_camera()
+
+    bgr_original = _capture_2d_image(camera)
+    cv2.imwrite("original.jpg", bgr_original)
+
+    print(f"Applying gamma correction: {user_options.gamma}")
+    bgr_adjusted = adjust_gamma(bgr_original, gamma=user_options.gamma)
```

Comment:
Not a comment but a question - what is the benefit of passing an argument as `gamma=user_options.gamma` instead of `user_options.gama`? Does it matter because the function is defined as `def adjust_gamma(image, gamma: float):`? What would be the difference if the function was defined as `def adjust_gamma(image, gamma):`?

Reply:
`gamma=` is required if there are default arguments in a function call, and you're using some defaults, but not all. I guess this is Python's overloading

Reply:
since we removed the default value we should probably just call it with positional arguments.